util-stats/src/main/scala/com/twitter/finagle/stats/InMemoryStatsReceiver.scala

```diff
@@ -47,8 +47,10 @@ object InMemoryStatsReceiver {
  * assert(isr.counters(Seq("a", "b", "foo") == 0)) // ok
  * assert(isr.counters(Seq("a", "b", "bar") == 0)) // fail
  * }}}
+ *
+ * @param maxStats the maximum number of stats, zero or negative value means unlimited (default is unlimited)
  **/
-class InMemoryStatsReceiver extends StatsReceiver with WithHistogramDetails {
+class InMemoryStatsReceiver(maxStats: Int = 0) extends StatsReceiver with WithHistogramDetails {
```

Comment:
instead of using a default arg, please overload the constructor so it's java friendly (java doesn't know how to deal with default args).

also, would it more clearly communicate our intent to pass this as an Option, and require that it's > 0 if specified?

Reply:
Yeah, so that we don't need to give a parameter in Java code.